src/commands/config.ts

```diff
@@ -1,85 +1,105 @@
 // Copyright (c) jdneo. All rights reserved.
 // Licensed under the GNU LGPLv3 license.
 
+import * as fse from 'fs-extra';
 import * as path from 'path';
-import { OpenDialogOptions, QuickPickItem, Uri, window, WorkspaceFolder } from 'vscode';
-import { BuiltinConfiguration } from '../constants/checkstyleConfigs';
-import { getDefaultWorkspaceFolder, setCheckstyleConfigurationPath } from '../utils/settingUtils';
+import { QuickPickItem, Uri, window, workspace, WorkspaceFolder } from 'vscode';
+import * as xmljs from 'xml-js';
+import { BuiltinConfiguration, DoctypePublicIds } from '../constants/checkstyleConfigs';
+import { asWorkspaceEnvPath as tryAsWorkspaceEnvPath, getDefaultWorkspaceFolder, setCheckstyleConfigurationPath } from '../utils/settingUtils';
 
-export async function setCheckstyleConfiguration(uri?: Uri): Promise<void> {
+export async function setConfiguration(uri?: Uri): Promise<void> {
     if (uri) {
         if (path.extname(uri.fsPath).toLowerCase() === '.xml') {
-            setCheckstyleConfigurationPath(uri.fsPath, uri);
+            setCheckstyleConfigurationPath(tryAsWorkspaceEnvPath(uri.fsPath), uri);
```

Comment:
What will happen if it's not a checkstyle xml?

Reply:
It only handles the path of xml file, the behavior is consistent with previous versions if the uri path isn't in workspace folder.